modules/core/client/components/ReadMorePanel.js

```diff
@@ -0,0 +1,50 @@
+// External dependencies
+import { withTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+
+// Internal dependencies
+import '@/config/client/i18n';
+import { plainText, plainTextLength } from '../utils/filters';
+
+const LIMIT = 2000;
+
+function ReadMorePanel({ content, id, t }) {
+  const [showMore, setShowMore] = useState(false);
+
+  if (content.length === 0) {
+    return null;
+  }
+
+  if (showMore || plainTextLength(content) <= LIMIT) {
+    return <div dangerouslySetInnerHTML={{ __html: content }} />;
+  }
+
+  return (
+    <div className="panel-more-wrap">
+      <div
+        className="panel-more-wrap"
+        dangerouslySetInnerHTML={{ __html: `${plainText(content).substr(0, LIMIT)} …` }}
```

Comment:
Good question! It's not desirable from a user experience / visual point of view. It's kinda mandatory if trying to avoid breaking HTML tags in the middle, tho. :-/

Consider e.g. breaking anywhere in the middle of `<a href="#">Hello</a>` and how destructive that might be. There could be some XSS type scenarios as well, although I can't think of any very harmful examples right now.

I'd like to somehow at least avoid breaking `<p><br/>` tho — I think that's avoidable quite simply.

That said, your suggestion to cut it with CSS is the "right" way to go about this: https://github.com/Trustroots/trustroots/pull/1161#pullrequestreview-338366175

> We could consider a different approach: Show the whole content, but wrap it in a block element with limited max-height and overflow: hidden. Then we would hide content based on content height, not its length.
>
> [This library](https://yarnpkg.com/en/package/react-overflow) could help with detecting whether the content overflows. However it has few people using it.

It started seeming a little more fiddly when considering different screen widths/heights/mobile affecting the panel width and calculating different height on each of those cases. It basically would need some kind of "max-height limit + overflow detector" to bail out from showing the "read more" if the text isn't overflowing, as you mention.

The actual logic for the component you linked is fairly simple, so just copying relevant bits to this component seems simple enough:

https://github.com/nickuraltsev/react-overflow/blob/b6ff220a23f60e84adf472882901eee6f149634a/src/OverflowDetector.jsx#L37-L48

Do you think it would be fine to merge this as is and loop back to bringing back the formatting within a few weeks?

Reply:
Good question! It's not desirable from a user experience / visual point of view. It's kinda mandatory if trying to avoid breaking HTML tags in the middle, tho. :-/

Consider e.g. breaking anywhere in the middle of `<a href="#">Hello</a>` and how destructive that might be. There could be some XSS type scenarios as well, although I can't think of any very harmful examples right now.

I'd like to somehow at least avoid breaking `<p><br/>` tho — I think that's avoidable quite simply.

That said, your suggestion to cut it with CSS is the "right" way to go about this: https://github.com/Trustroots/trustroots/pull/1161#pullrequestreview-338366175

> We could consider a different approach: Show the whole content, but wrap it in a block element with limited max-height and overflow: hidden. Then we would hide content based on content height, not its length.
>
> [This library](https://yarnpkg.com/en/package/react-overflow) could help with detecting whether the content overflows. However it has few people using it.

It started seeming a little more fiddly when considering different screen widths/heights/mobile affecting the panel width and calculating different height on each of those cases. It basically would need some kind of "max-height limit + overflow detector" to bail out from showing the "read more" if the text isn't overflowing, as you mention.

The actual logic for the component you linked is fairly simple, so just copying relevant bits to this component seems simple enough:

https://github.com/nickuraltsev/react-overflow/blob/b6ff220a23f60e84adf472882901eee6f149634a/src/OverflowDetector.jsx#L37-L48

Do you think it would be fine to merge this as is and loop back to bringing back the formatting within a few weeks?

Reply:
> Do you think it would be fine to merge this as is and loop back to bringing back the formatting within a few weeks?

If we merge to production in the meantime, users will be confused. Otherwise it's fine I think. So I guess decide based on the production plans... :wink: